src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -1626,10 +1626,20 @@ sealed class RsDiagnostic(
             fixes = listOf(AddAssocTypeBindingsFix(element, missingTypes.map { it.name }))
         )
     }
+
+    class ConstOrTypeParamsInExternError(
+        element: PsiElement
+    ) : RsDiagnostic(element) {
+        override fun prepare() = PreparedAnnotation(
+            ERROR,
+            E0044,
+            "Foreign items may not have type parameters"
```

Comment:
@vlad20012 what do you think about the extraction of all new diagnostic errors into message bundle (probably separate new one) not to produce more code that we need to refactor in the future anyway for localization purposes?